protocol/flows/src/v5/ibd/flow.rs

```diff
@@ -107,6 +107,16 @@ impl IbdFlow {
                 .await?;
             }
             IbdType::DownloadHeadersProof => {
+                if unix_now() - session.async_creation_timestamp().await > self.ctx.config.finality_duration() {
```

Comment:
1. Better to do `unix_now() > self.ctx.config.finality_duration() + session.async_creation_timestamp().await` in order to avoid even a theoretical underflow  
2. Same goes for line 113
3. I think this check should be part of `determine_ibd_type` (inserted at line 168; returning `IbdType::None` if needed)
4. Missing a comment explaining this condition. "We reject the headers proof if the node has a relatively up-to-date finality point and current consensus has matured for long enough (and not recently synced). This is mostly a spam-protector since subsequent checks identify these violations as well"